Kong/docs.konghq.com
# feat: Konnect API examples for plugins

### Description

For any plugin that is supported in Konnect, it’s now possible to configure it using the Konnect runtime groups API, via `core-entities`. This PR adds support for generated Konnect API examples in the plugin docs, which makes the Konnect API more visible and positions it as a parallel option to the Kong Admin API.

A simple service example for the Basic auth plugin:
![Screenshot 2023-08-17 at 1 54 20 PM](https://github.com/Kong/docs.konghq.com/assets/54370747/2168aff3-0bad-4d46-82f9-ed5964a083c2)

A global example for the Rate Limiting Advanced plugin:
![Screenshot 2023-08-17 at 2 57 19 PM](https://github.com/Kong/docs.konghq.com/assets/54370747/d71f434e-7689-4a7a-bad4-9298621e49eb)

Along with that, there were several fixes/improvements that I had to make:
* Placeholders in curl examples: after team discussion, settled on `{ }` as the placeholder characters for URLs. This format is heavily used in our API specs, and is the OpenAPI standard: https://swagger.io/docs/specification/describing-parameters/. Going forward, we should be using `{ }` to enclose variables **in specs and API examples**. 
  * Placeholders in other situations still need revisiting. Out of scope here.
 
* The global example now explains what "global" means in the context of Konnect, Kong Enterprise, and Kong OSS. This is a long-running painpoint, and it seems timely to add this as we add Konnect examples.

* Removed the duplicate `konnect_examples` and `manager_examples` metadata keys from plugin docs. We don't have Kong Manager examples, and we don't need a separate `konnect_examples` key. `konnect: false/true` already performs the same function.

https://konghq.atlassian.net/browse/DOCU-2949

### Testing instructions

Check any plugin doc. For example:
* [Rate Limiting Advanced plugin](https://deploy-preview-5970--kongdocs.netlify.app/hub/kong-inc/rate-limiting-advanced/how-to/basic-example/) - has all example types, good way to see everything in one place
* [OAuth2 plugin](https://deploy-preview-5970--kongdocs.netlify.app/hub/kong-inc/oauth2/how-to/basic-example/) - isn't supported in Konnect so the Konnect API tab doesn't show up
* [Update to style guide](https://deploy-preview-5970--kongdocs.netlify.app/contributing/style-guide/#placeholder-and-example-values)


### Checklist
